CDAP-13853 remove data pipeline requirement for a time schedule

albertshau opened the pull request:
Time schedule is no longer a requirement to create a batch
pipeline. Schedules can now be created after an app is deployed,
so there is no need to supply it at configure time.

albertshau committed:
CDAP-13853 remove data pipeline requirement for a time schedule

Time schedule is no longer a requirement to create a batch
pipeline. Schedules can now be created after an app is deployed,
so there is no need to supply it at configure time.

anew commented on cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/co/cask/cdap/etl/proto/v2/ETLBatchConfig.java:
```diff
@@ -141,10 +142,20 @@ public String toString() {
 
+  /**
+   * @return a builder used to create the config for a data pipeline
+   */
```
If the schedule is optional, then it seems that the pattern should be:
```
... = builder().
   ...
   .addTimeSchedule()
   ...
   .build(); 
```
Other than backward-compatibility, is there a reason not to have it that way?

albertshau replied:
true, I'll add a `setTimeSchedule` (multiple time schedules are not supported).

anew replied:
👍

anew commented on cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java:
```diff
@@ -3238,7 +3237,7 @@ private void testSplitterToJoiner(Engine engine) throws Exception {
      * signups --> namesplitter --|
      *                            |non-null --> sink2
      */
-    ETLBatchConfig config = ETLBatchConfig.builder("* * * * *")
```
hard to see from the diff: did you keep the schedule for a few of the test cases?

albertshau replied:
not in these tests that run actual pipelines, since none of these tests use a time schedule. It is still used in some other tests that test planning and upgrade.

albertshau committed:
add a setTimeSchedule() method to pipeline config builder

Adding a builder method to set a time schedule and deprecating
the old method that passes time schedule through a constructor.

anew reviewed (approved):
LGTM

albertshau commented:
https://builds.cask.co/browse/CDAP-DUT6569
